pkg/controller/queuejob/utils.go

```diff
@@ -95,6 +95,11 @@ func createQueueJobPod(qj *arbv1.QueueJob, template *corev1.PodTemplateSpec, ix
 		},
 		Spec: templateCopy.Spec,
 	}
+	// we replace the scheduler name (if any) specified in the pod definition with the one specified in the QJ template
+	if qj.Spec.SchedulerName != "" {
```

Comment:
It's better to check Pod's scheduler name: if it's empty, replace it with QueueJob's schedule name.

Reply:
@k82cn : I've also added a check for the schedulerName in the podTemplate based on your comment.